### ISSUE TITLE
Image calssification for 10 classes

### ISSUE THREAD

**Issue by omair97** [HamadYA/GhostFaceNets#49]
HI i hope you are doing fine. I wanted to use these weights for classification. can you provide simple code for that? 
what i have to add at the last layer for classification

**Comment by HamadYA** [comment]
Hi,
You simply need to replace the last layer with a fully connected layer matching the number of your classes with softmax activation (for multi-class) or sigmoid (for binary classification).

A simple code would be as below.
I would recommend to use GhostNet implementation and use it as a feature extractor because GhostFaceNets is only for face recognition. You may check this: https://github.com/leondgarse/keras_cv_attention_models/tree/main/keras_cv_attention_models/ghostnet

import torch
import torch.nn as nn

class FaceRecognitionClassifier(nn.Module):
    def __init__(self, feature_extractor, num_classes):
        super(FaceRecognitionClassifier, self).__init__()
        self.feature_extractor = feature_extractor  # Pretrained face recognition model
        self.fc = nn.Linear(feature_extractor.output_dim, num_classes)  # Adjust output_dim as needed

    def forward(self, x):
        features = self.feature_extractor(x)  # Extract features
        logits = self.fc(features)  # Classification head
        return logits

num_classes = 10  # Change as needed
pretrained_model = ...  # Load your face recognition model
classifier_model = FaceRecognitionClassifier(pretrained_model, num_classes)

**Comment by HamadYA** [comment]
> Hi, You simply need to replace the last layer with a fully connected layer matching the number of your classes with softmax activation (for multi-class) or sigmoid (for binary classification).
> 
> A simple code would be as below. I would recommend to use GhostNet implementation and use it as a feature extractor because GhostFaceNets is only for face recognition. You may check this: https://github.com/leondgarse/keras_cv_attention_models/tree/main/keras_cv_attention_models/ghostnet
> 
> import torch import torch.nn as nn
> 
> class FaceRecognitionClassifier(nn.Module): def **init**(self, feature_extractor, num_classes): super(FaceRecognitionClassifier, self).**init**() self.feature_extractor = feature_extractor # Pretrained face recognition model self.fc = nn.Linear(feature_extractor.output_dim, num_classes) # Adjust output_dim as needed
> 
> ```
> def forward(self, x):
>     features = self.feature_extractor(x)  # Extract features
>     logits = self.fc(features)  # Classification head
>     return logits
> ```
> 
> num_classes = 10 # Change as needed pretrained_model = ... # Load your face recognition model classifier_model = FaceRecognitionClassifier(pretrained_model, num_classes)

**Comment by HamadYA** [comment]
I am closing this issue, if you have more questions feel free to reopen it.
